packages/video_player/video_player/android/src/test/java/io/flutter/plugins/videoplayer/VideoPlayerTest.java

```diff
@@ -4,12 +4,72 @@
 
 package io.flutter.plugins.videoplayer;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.android.exoplayer2.Format;
+import com.google.android.exoplayer2.SimpleExoPlayer;
+import io.flutter.plugin.common.EventChannel;
+import io.flutter.view.TextureRegistry;
+import java.util.Map;
 import org.junit.Test;
+import org.mockito.ArgumentCaptor;
 
 public class VideoPlayerTest {
   // This is only a placeholder test and doesn't actually initialize the plugin.
   @Test
   public void initPluginDoesNotThrow() {
     final VideoPlayerPlugin plugin = new VideoPlayerPlugin();
   }
+
+  @Test
+  public void videoPlayerSendInitializedSetsRotationCorrectionForRotationDegrees180() {
+    Format format = new Format.Builder().setRotationDegrees(180).build();
+    SimpleExoPlayer mockExoPlayer = mock(SimpleExoPlayer.class);
+    when(mockExoPlayer.getVideoFormat()).thenReturn(format);
+    final VideoPlayer player =
+        new VideoPlayer(
+            mock(EventChannel.class),
+            mock(TextureRegistry.SurfaceTextureEntry.class),
+            mock(VideoPlayerOptions.class),
+            mockExoPlayer);
+    QueuingEventSink mockEventSink = mock(QueuingEventSink.class);
+    player.eventSink = mockEventSink;
+    player.isInitialized = true;
+
+    player.sendInitialized();
```

Comment:
Why do you need to use this internal method? You're mocking the exoplayer that has a listener set on it, which means you should be able to capture the listener and invoke the correct state change on it to drive this flow rather than exposing several pieces of internals. That would make the test much less fragile.

Reply:
I reverted the tests. (See https://github.com/flutter/plugins/pull/3820#issuecomment-1086419754)